### PR TITLE
add AOP hyperspectral reflectance dataset descriptions

### PR DESCRIPTION
**Pull request by bhass-neon**
Catalog and examples for NEON AOP surface directional reflectance. Had to add some units to eo_extensions.py for unit-related tests to pass.

### CODE REVIEW

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 1]
```diff
@@ -0,0 +1,3859 @@
...
+      'GAO',
+      'NIS1', 
+      'NIS2',
+      'NIS3',
+    ],
+    'eo:bands': [
+     {
+      name: "B001",
+      description: "VNIR Band B001",
+      'gee:units': 'unitless',
```
Comment (simonff) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:154]: We don't typically say "unitless" for such bands... though Hyperion's units are  'W/m^2 SR&mu;m',

Have you see reflectance units be specified in other places?

File: examples/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+var refl001 = ee.ImageCollection(
+  "projects/neon-prod-earthengine/assets/HSI_REFL/001");
```
Comment (simonff) [inline examples/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.js:2]: asset not publicly readable

Comment (bhass-neon) [reply examples/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.js:2]: Fixed this

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 3]
```diff
@@ -0,0 +1,3426 @@
...
+      ]
+     },
+     {
+      name: 'B432',
+      'gee:units': 'deg',
+      description: 'Illumination Factor used as input to ATCOR'
+     },
+     {
+      name: 'B433',
+      'gee:units': 'm',
```
Comment (bhass-neon) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3311]: Should we use units.meter, like https://github.com/NEONScience/earthengine-catalog/blob/dd5e2037ef00cfe7ef93ff86cf638ee41bacaac7/catalog/USGS/USGS_NED.jsonnet#L67 ?

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 4]
```diff
@@ -0,0 +1,3859 @@
...
+    Guide](https://data.neonscience.org/api/v0/documents/quick-start-guides/NEON.QSG.DP3.30006.001v1?inline=true&fallback=html)
+  |||,
+  license: license.id,
+  links: ee.standardLinks(subdir, id),
+  keywords: [
+#    'airborne',
+    'canopy',
+#    'ecology',
+    'forest',
+#    'hyperspectral',
```
[the remark this answers is not in the document]
Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet]: I am adding back in some of these commented out keywords. The checks are failing with the message `Only one instance of ...`. Is it ok to leave these un-commented with the understanding that once all datasets are added, they will pass?

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 5]
```diff
@@ -0,0 +1,3426 @@
...
+  links: ee.standardLinks(subdir, id),
+  keywords: [
+    'airborne',
+    'canopy',
+    'ecology',
+    'forest',
+    'hyperspectral',
+    'neon',
+    'reflectance',
+    'surface-reflectance',
```
Comment (bhass-neon) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:50]: the `surface-reflectance` keyword is failing with this message: "'keyword must contain only lowercase letters, digits, and underscores and be at most 49 characters long: "surface-reflectance"', IssueLevel.error)", as well as an "only one instance of" error, but I see it in the GEE dataset tags here: https://developers.google.com/earth-engine/datasets/tags/surface-reflectance.

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 6]
```diff
@@ -0,0 +1,3427 @@
...
+    
+    Documentation: [NEON DP3.30006.001 Spectrometer orthorectified surface directional reflectance - mosaic Quick Start
+    Guide](https://data.neonscience.org/api/v0/documents/quick-start-guides/NEON.QSG.DP3.30006.001v1?inline=true&fallback=html)
+  |||,
+  license: license.id,
+  links: ee.standardLinks(subdir, id),
+  keywords: [
+    'airborne',
+    'canopy',
+    'ecology',
```
Comment (simonff) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:47]: Recommend removing 'ecology' as very broad.

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 7]
```diff
@@ -0,0 +1,3427 @@
+local id = 'projects/neon-prod-earthengine/assets/HSI_REFL/001';
+local subdir = 'neon-prod-earthengine';
+local ee_const = import 'earthengine_const.libsonnet';
+local ee = import 'earthengine.libsonnet';
+local spdx = import 'spdx.libsonnet';
+local license = spdx.proprietary;
```
Comment (simonff) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:6]: if the license is pure CC 1.0, please use
local license = spdx.cc0_1_0;

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 8]
```diff
@@ -0,0 +1,3433 @@
...
+      name: "B426",
+      description: "SWIR Band B426",
+      center_wavelength: 2.5109500885,
+      full_width_half_max: 0.0064290729,
+      'gee:wavelength': '2510.9507nm, FWHM: 6.4291nm',
+     },
+     {
+      name: 'B427',
+      'gee:units':'deg',
+      description: 'Aerosol Optical Depth'
```
Comment (dpencosk) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3136]: Use units.degree from units.libsonnet

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 9]
```diff
@@ -0,0 +1,3433 @@
...
+     },
+     {
+      name: 'B428',
+      'gee:units': 'deg',
```
Comment (dpencosk) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3140]: Also here

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet [hunk 10]
```diff
@@ -0,0 +1,3433 @@
...
+            value: 2,
+            color: 'c40a0a',
+            description: 'mostly cloudy: >50% cloud cover',
+            },
+      ],
+      description: 'Weather Quality Indicator - estimated percentage of overhead cloud cover during acquisition'
+     },
+     {
+      name: 'B442',
+//      'gee:units': 'date yyyy-mm-dd',
```
Comment (dpencosk) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3379]: No unit needed for this band

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet [hunk 11]
```diff
@@ -0,0 +1,3431 @@
...
+     {
+      name: "B426",
+      description: "SWIR Band B426",
+      center_wavelength: 2.5109500885,
+      full_width_half_max: 0.0064290729,
+      'gee:wavelength': '2510.9507nm, FWHM: 6.4291nm',
+     },
+     {
+      name: 'B427',
+      'gee:units':'deg',
```
Comment (dpencosk) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet:3133]: units.degree here

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet [hunk 12]
```diff
@@ -0,0 +1,3431 @@
...
+            value: 2,
+            color: 'c40a0a',
+            description: 'mostly cloudy: >50% cloud cover',
+            },
+      ],
+      description: 'Weather Quality Indicator - estimated percentage of overhead cloud cover during acquisition'
+     },
+     {
+      name: 'B442',
+//      'gee:units': 'date yyyy-mm-dd',
```
Comment (dpencosk) [inline catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet:3377]: Can remove unit here

File: checker/node/eo_extension.py [hunk 13]
```diff
@@ -228,6 +228,7 @@
     'molec cm-2 s-1',
     'ms',
     'occurrence',
+    'percent',
```
Comment (dpencosk) [inline checker/node/eo_extension.py:231]: This should be unnecessary since you're using units.percent

**Review by dpencosk — CHANGES_REQUESTED** [review]
Looks really good! A few small nitpicks are all I could find.

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet (hunk 11 above)
Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet:3133]: checking that the latest version is up. I updated all these units to match what's in the 001 catalog

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet (hunk 8 above)
Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3136]: see https://github.com/google/earthengine-catalog/pull/1058/commits/b94ae22110e294378dd771521a6105e68c823d9e - I think what you're looking at is dated

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet (hunk 11 above)
Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet:3133]: see comment above & https://github.com/google/earthengine-catalog/pull/1058/commits/b94ae22110e294378dd771521a6105e68c823d9e

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet (hunk 8 above)
Comment (dpencosk) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3136]: The changes were made to bands 432 onward, but they're missing from 427 and 428.

Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3136]: Ah, sorry I missed this, just fixed in: https://github.com/google/earthengine-catalog/pull/1058/commits/243970724367580ce1b059d3276e8315342a5f93

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet (hunk 11 above)
Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet]: resolved in https://github.com/google/earthengine-catalog/pull/1058/commits/243970724367580ce1b059d3276e8315342a5f93

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet (hunk 10 above)
Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet]: resolved in https://github.com/google/earthengine-catalog/pull/1058/commits/32addbee9bc71da76b33d5aac3f8a198263e491b

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet (hunk 12 above)
Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_002.jsonnet]: resolved in https://github.com/google/earthengine-catalog/pull/1058/commits/32addbee9bc71da76b33d5aac3f8a198263e491b

**Comment by bhass-neon** [comment]
@dpencosk I think I fixed all your catches, thanks!

File: catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet (hunk 8 above)
Comment (dpencosk) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3136]: They're fixed in 002 now but the same bands in 001 are still 'deg'

Comment (bhass-neon) [reply catalog/neon-prod-earthengine/projects_neon-prod-earthengine_assets_HSI_REFL_001.jsonnet:3136]: Oops! Just fixed here: https://github.com/google/earthengine-catalog/pull/1058/commits/f92f024443cfc7f369055247aec4edc1928528b8

**Pull request merged**